File: leetcode_go/problems/0219_contains_duplicate_ii/contains_duplicate_ii.go
package problem0219

import "math"

func containsNearbyDuplicate(nums []int, k int) bool {
	dict := make(map[int]int)

	for i, n := range nums {
		if j, ok := dict[n]; ok {
			if i != j && int(math.Abs(float64(i-j))) <= k {
				return true
			}
		}

		dict[n] = i
	}

	return false
}

File: leetcode_python/lib/problems/p0219_contains_duplicate_ii/contains_duplicate_ii.py
from typing import List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        dictionary = {}
        for i in range(len(nums)):
            n = nums[i]
            j = dictionary.get(n)

            if j != None and i != j and abs(i - j) <= k:
                return True

            dictionary[n] = i

        return False
